drivers/pwm/pwm_stm32.c

```diff
@@ -140,20 +119,16 @@ static int pwm_stm32_get_cycles_per_sec(struct device *dev, u32_t pwm,
 {
 	const struct pwm_stm32_config *cfg = DEV_CFG(dev);
 	struct pwm_stm32_data *data = DEV_DATA(dev);
-	u32_t bus_clk, tim_clk;
+	u32_t bus_clk;
 
 	if (cycles == NULL) {
 		return -EINVAL;
 	}
 
-	/* Timer clock depends on APB prescaler */
 	clock_control_get_rate(data->clock,
 			(clock_control_subsys_t *)&cfg->pclken, &bus_clk);
 
-	tim_clk = __get_tim_clk(bus_clk,
-			(clock_control_subsys_t *)&cfg->pclken);
-
-	*cycles = (u64_t)(tim_clk / (data->pwm_prescaler + 1));
+	*cycles = (u64_t)(bus_clk / (data->pwm_prescaler + 1));
```

Comment:
Found out what was causing my confusion, there is nothing saying that the timer clock needs to take the APB prescale into account in the STM32F401RE manual (only "The timers connected to APB2 are clocked from TIMxCLK up to 84 MHz, while the timers connected to APB1 are clocked from TIMxCLK up to 42 MHz").

Now checking at the other F/L manuals, they all state what you just described above. Also took a look at stm32cube for F4xx and the same sentence is covered in the pwm sample readme.txt.